### PR TITLE
crud: support load balancer work requests; general cleanup

### PR DESCRIPTION
**Pull request by josephholsten**

### CODE REVIEW

File: crud/helpers.go [hunk 1]
```diff
@@ -38,17 +36,24 @@ func (s *BaseCrud) VoidState() {
...
+					currentState := stateValue.String()
+					log.Printf("[DEBUG] crud.BaseCrud.setState: state: %#v", currentState)
+					return s.D.Set("state", currentState)
+				}
 			}
-			return nil
 		}
 	}
 
+	// TODO: should this be an error?
```
Comment (mikejihbe) [inline crud/helpers.go]: Probably. Maybe even panic because this indicates a developer error, right?

File: crud/helpers.go [hunk 2]
```diff
@@ -162,7 +176,14 @@ func waitForStateRefresh(sync StatefulResource, timeout time.Duration, pending,
 		handleMissingResourceError(sync, &e)
 		return
 	}
-	sync.SetData()
 
 	return
 }
+
+func FilterMissingResourceError(sync ResourceVoider, err *error) {
```
Comment (mikejihbe) [inline crud/helpers.go]: Not seeing this used anywhere. Should it be? If not it can probably be removed?

File: crud/interface.go [hunk 3]
```diff
@@ -10,6 +10,9 @@ import (
 
 // Gets the current BareMetal Resource
 type ResourceFetcher interface {
+	// Get should update the s.Resource, and is used by crud.ReadResource() to populate s.D
+	// Get() may expect s.D.Id() to be set, but not s.Resource, or anything else in s.D
+	// TODO: switch to expect s.ID() instead of s.D.Id()
```
Comment (mikejihbe) [inline crud/interface.go]: This seems like a big TODO and is a little worrisome. Let's chat about it.

File: crud/interface.go [hunk 4]
```diff
@@ -48,6 +52,9 @@ type ResourceDeleter interface {
 	Delete() error
 }
 
+// }}}
```
Comment (mikejihbe) [inline crud/interface.go]: Extraneous comment

File: crud/interface.go [hunk 5]
```diff
@@ -48,6 +52,9 @@ type ResourceDeleter interface {
...
+
+// TODO: doc
```
Comment (mikejihbe) [inline crud/interface.go]: "Some resources in the oracle API are removed asynchronously, so even after they claim to be gone, other dependencies haven't been notified of that fact. This facility allows us to add an artificial delay for resources that need a little time to let the oracle API backend catch up with reality."

**Review by mikejihbe — COMMENTED** [review]
Generally looks good. I made a few minor nit-picks

File: crud/interface.go (hunk 3 above)
Comment (josephholsten) [reply crud/interface.go]: This is totally incorrect, and was based on my misunderstanding the point of s.ID().

File: crud/helpers.go (hunk 2 above)
Comment (josephholsten) [reply crud/helpers.go]: ah yes. This was just a public version of `handleMissingResourceError()` for the #15 spike

File: crud/interface.go (hunk 5 above)
Comment (josephholsten) [reply crud/interface.go]: +900

Comment (josephholsten) [reply crud/interface.go]: or 9000

**Pull request merged**